lldb/source/Core/PluginManager.cpp

```diff
@@ -705,14 +705,18 @@ Status PluginManager::SaveCore(const lldb::ProcessSP &process_sp,
   Status error;
   auto &instances = GetObjectFileInstances().GetInstances();
   for (auto &instance : instances) {
-    if (plugin_name.empty() || instance.name == plugin_name) {
-      if (instance.save_core &&
-          instance.save_core(process_sp, outfile, core_style, error))
+    if (options.GetCoreDumpPluginName()->empty() ||
```

Comment:
Use the optional `plugin_name` from above:
```
if (!plugin_name.has_value() || instance.name == plugin_name.value()) {
```

Reply:
This will crash of the plug-in name has no value. Add a test for this as mentioned above. See above comment out "Get the plugin optional here outside of the loop:" and implement the fix suggested above to avoid crashing due to an STL exception